### PR TITLE
Feature/icon component

### PR DESCRIPTION
**Pull request by RubenDeAndrade**
I created an Icon component.
It has a dictionnary with all the icons it can supports. You just need to give him the right icon name.
For now, it only supports one Icon: the search Icon;

# Tests

Go to storybook: `yarn storybook`

### CODE REVIEW

File: src/stories/Icon/Icon.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import { ReactComponent as SearchIcon } from './iconSvgs/search.svg';
+import './icon.css';
+
+type IconKey = 'search';
+
+type Dictionary<T> = {
+    [Key in IconKey]: T;
+};
+
+type Svg = React.FunctionComponent<React.SVGProps<SVGSVGElement> & {
```
Comment (1Green) [inline src/stories/Icon/Icon.tsx]: Besoin du `title` ?

File: src/stories/Icon/Icon.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    title?: string | undefined;
+}>
+
+const icons: Dictionary<Svg> = {
```
Comment (1Green) [inline src/stories/Icon/Icon.tsx]: Tu pourrais te passer de tes types intermédiaires et juste écrire 
```Typescript
type IconKey = 'search'
type Icons =  { [key in IconKeys]: FC<SVGProps<SVGSVGElement>> }
const icons: Icons = {}
```

File: src/stories/Icon/iconSvgs/search.svg [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	 viewBox="0 0 512.005 512.005" style="enable-background:new 0 0 512.005 512.005;" xml:space="preserve">
+<g>
+	<g>
+		<path d="M505.749,475.587l-145.6-145.6c28.203-34.837,45.184-79.104,45.184-127.317c0-111.744-90.923-202.667-202.667-202.667
+			S0,90.925,0,202.669s90.923,202.667,202.667,202.667c48.213,0,92.48-16.981,127.317-45.184l145.6,145.6
+			c4.16,4.16,9.621,6.251,15.083,6.251s10.923-2.091,15.083-6.251C514.091,497.411,514.091,483.928,505.749,475.587z
+			 M202.667,362.669c-88.235,0-160-71.765-160-160s71.765-160,160-160s160,71.765,160,160S290.901,362.669,202.667,362.669z"/>
+	</g>
+</g>
+<g>
```
Comment (1Green) [inline src/stories/Icon/iconSvgs/search.svg]: Pas très propre ce SVG :) Plein de groupes vides.

File: src/stories/Icon/Icon.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    className,
+    size = 'medium',
+    ...rest
+}) => {
+    const CustomIcon = icons[name];
+
+    return (
+        <CustomIcon
+            fill={color}
+            className={[className, `icon--${size}`].join(' ')}
```
Comment (victorchabbert) [inline src/stories/Icon/Icon.tsx]: utiliser `classnames` partout.

File: src/stories/Icon/iconSvgs/search.svg [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
```
Comment (victorchabbert) [inline src/stories/Icon/iconSvgs/search.svg]: Passer svgo

**Review by victorchabbert — APPROVED** [review]

**Review by williamgoulois — APPROVED** [review]

**Pull request merged**